apps/chat-e2e/src/tests/promptFoldersSpecialChars.test.ts

```diff
@@ -0,0 +1,225 @@
+    const newNameWithSpecialChars = `${folderName} ${ExpectedConstants.allowedSpecialChars}`;
+    const nameWithRestrictedChars = `Folder=,:;{}\\/%&"_name`;
+    const expectedFolderName = 'Folder_name';
+    const newNameWithSpaces = 'Folder   1';
+    const nameWithSpacesBeforeAndAfter = '   Folder with spaces   ';
+    const expectedName = 'Folder with spaces';
+    const newNameWithEmojis = '😂👍🥳 😷 🤧 🤠 🥴😇 😈 ⭐あおㅁㄹñ¿äß';
+
+    await dialTest.step('Create prompt folder', async () => {
+      await dialHomePage.openHomePage();
+      await dialHomePage.waitForPageLoaded();
+
+      await promptBar.createNewFolder();
+      await expect
+        .soft(
+          folderPrompts.getFolderByName(folderName),
+          ExpectedMessages.folderIsVisible,
+        )
+        .toBeVisible();
+    });
+
+    await dialTest.step(
+      'Rename it to have a dot at the end of the name',
+      async () => {
+        await folderPrompts.openFolderDropdownMenu(folderName);
+        await folderDropdownMenu.selectMenuOption(MenuOptions.rename);
+        await folderPrompts.editFolderName(newNameWithEndDot);
+      },
+    );
+
+    await dialTest.step('Click on confirmation button', async () => {
+      await folderPrompts.getEditFolderInputActions().clickTickButton();
+
+      const errorMessage = await errorToast.getElementContent();
+      expect
+        .soft(errorMessage, ExpectedMessages.notAllowedNameErrorShown)
+        .toBe(ExpectedConstants.nameWithDotErrorMessage);
+
+      // Verify folder name stays in edit mode
+      await expect
+        .soft(
+          folderPrompts.getEditFolderInput().getElementLocator(),
+          ExpectedMessages.folderEditModeIsActive,
+        )
+        .toBeVisible();
+    });
+
+    // Wait for the toast message to disappear
+    await errorToast.getElementLocator().waitFor({ state: 'hidden' });
+
+    await dialTest.step('Rename it to contain special characters', async () => {
+      await folderPrompts.editFolderNameWithTick(newNameWithSpecialChars);
+      await expect
+        .soft(
+          folderPrompts.getFolderByName(newNameWithSpecialChars),
+          ExpectedMessages.folderNameUpdated,
+        )
+        .toBeVisible();
+    });
+
+    await dialTest.step(
+      'Try to type restricted special characters',
+      async () => {
+        await folderPrompts.openFolderDropdownMenu(newNameWithSpecialChars);
+        await folderDropdownMenu.selectMenuOption(MenuOptions.rename);
+        for (const char of ExpectedConstants.restrictedNameChars.split('')) {
+          await folderPrompts.editFolderName(char);
+          expect
+            .soft(
+              await folderPrompts.getEditFolderInput().getEditInputValue(),
+              ExpectedMessages.charactersAreNotDisplayed,
+            )
+            .toBe('');
+        }
+      },
+    );
+
+    await dialTest.step(
+      'Copy and paste restricted special characters',
+      async () => {
+        await page.evaluate(
```

Comment:
let's move this useful method to BasePage class, because it can be reused in other tests

Reply:
done